db/pebble/db_test.go

```diff
@@ -221,3 +222,94 @@ func TestConcurrentUpdate(t *testing.T) {
 		})
 	}))
 }
+
+func TestSeek(t *testing.T) {
+	testDb := pebble.NewMemTest()
+	defer testDb.Close()
+
+	err := testDb.Update(func(txn db.Transaction) error {
+		err := txn.Set([]byte{1}, []byte{1})
+		assert.NoError(t, err)
+		err = txn.Set([]byte{3}, []byte{3})
+		assert.NoError(t, err)
+		return nil
+	})
+	assert.NoError(t, err)
+
+	testDb.View(func(txn db.Transaction) error {
+		t.Run("seeks to the next key in lexicographical order", func(t *testing.T) {
+			iter, _ := txn.NewIterator()
+			defer iter.Close()
+
+			iter.Seek([]byte{0})
+			assert.Equal(t, []byte{1}, iter.Key())
+		})
+
+		t.Run("key returns nil when seeking nonexistent data", func(t *testing.T) {
+			iter, _ := txn.NewIterator()
+			defer iter.Close()
+
+			iter.Seek([]byte{4})
+			assert.Nil(t, iter.Key())
+		})
+		return nil
+	})
+}
+
+type Entry struct {
```

Comment:
I dont think this is used anymore. There is also another `Entry` in `db.go`, we can remove both of them

Reply:
It is actually being used [here](https://github.com/NethermindEth/juno/pull/572/files#diff-e9f741c3b54d069564c343dbbe5833e0e95207c50232497ce32d9a1decc31a07R296-R302). I've removed the one in `db.go`